bqplot/traits.py

```diff
@@ -145,6 +145,8 @@ def _set_value(self, value):
     def set(self, obj, value):
         #TODO: We shouldnt have to overload the set because
         # numpy doesn't support == comparions
+        #TODO: This should also be removed once _buffered_state_diff is
+        # fixed in ipywidgets
```

Comment:
Well, previously, `_buffered_state_diff` only returned what trait changed. Now for the first two calls it returns everything. 

The behavior of `_buffered_state_diff` should be consistent with every call I feel. It should either return everything always or return only the changed things but not change depending on if it is the first call or the fourth.